### PR TITLE
feat(core): Add all currencies in Currency Enum 

### PR DESCRIPTION
**Pull request by tusharxoxoxo**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

### Additional Changes

- [x] This PR modifies the API contract
- [x] This PR modifies the database schema
- [x] This PR modifies application configuration/environment variables

## Motivation and Context
#1814 

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [ ] I addressed lints thrown by `cargo clippy`
- [x] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

**Comment by tusharxoxoxo** [comment]
`cargo clippy` ran perfectly fine
image
`cargo +nightly fmt` also ran perfectly fine
Screenshot 2023-08-02 at 2 59 24 AM

But after running 
`cargo check --all-features
cargo clippy --all-features
cargo test --all-features`
there were some compilation errors
Screenshot 2023-08-02 at 2 57 52 AM
Screenshot 2023-08-02 at 2 58 19 AM
Screenshot 2023-08-02 at 2 58 35 AM
Screenshot 2023-08-02 at 2 58 57 AM

**Comment by pixincreate** [comment]
Hey @tusharxoxoxo, try running migrations, then perform a `cargo clean` once. Try compiling them again after that.

**Comment by tusharxoxoxo** [comment]
I don't know how to run migrations, can u please tell me?
I have asked the very same question on the discord server of hyperswitch
today

**Comment by prasunna09** [comment]
Hey Hi, did the previous error `linking with 'cc' failed` got fixed?

File: migrations/2023-07-19-081050_add_zero_decimal_currencies/up.sql [hunk 1]
```diff
@@ -1,14 +1,75 @@
 -- Your SQL goes here
+ALTER TYPE "Currency" ADD VALUE IF NOT EXISTS 'AFN' AFTER 'AED';
```
Comment (prasunna09) [inline migrations/2023-07-19-081050_add_zero_decimal_currencies/up.sql:2]: Here you cannot edit the old file. Instead you have to create new one.
Use this command. The files will be automatically created - `diesel migration generate <migration_file_name>`.
Two files will be created - `up.sql` should have all these queries
`down.sql` should have `SELECT 1;`

**Review by prasunna09 — CHANGES_REQUESTED** [review]

**Comment by uddhavdave** [comment]
@tusharxoxoxo the error logs show `library pq not found` , can you try running the steps mentioned [here](https://github.com/juspay/hyperswitch/blob/main/docs/try_local_system.md#set-up-dependencies-on-macos).

>If linking diesel_cli fails due to missing libpq (if the error message is along the lines of cannot find -lpq), you may also have to install libpq and reinstall diesel_cli:
> ```
> brew install libpq
> export PQ_LIB_DIR="$(brew --prefix libpq)/lib"
> 
> cargo install diesel_cli --no-default-features --features "postgres"
> ```
> You may also choose to persist the value of PQ_LIB_DIR in your shell startup file like so:
> ```
> echo 'PQ_LIB_DIR="$(brew --prefix libpq)/lib"' >> ~/.zshrc
> ```

**Pull request closed without merging**